core/v1/board.cpp

```diff
@@ -115,84 +127,81 @@ void PBoard::floodfill(Combo *list, const OrbLocation &loc, const Orb &orb, int
+            for (int i = startCount; i < endCount; i++)
             {
                 int cx = x;
                 int cy = y;
                 if (d == 0)
                     cy += i;
                 else if (d == 1)
-                    cy -= i;
-                else if (d == 2)
                     cx += i;
-                else if (d == 3)
-                    cx -= i;
 
-                // only need to check two orbs because by default, it needs to have at least 3 connected orbs to be considered as a combo
-                bool hasOrbAround = false;
-                if (d < 2)
+                int index = INDEX_OF(cx, cy);
+                if (temp[index] == 0)
                 {
-                    // check up down
-                    hasOrbAround = hasSameOrb(orb, OrbLocation(cx + 1, cy)) && hasSameOrb(orb, OrbLocation(cx - 1, cy));
+                    board[index] = pad::empty;
+                    list->emplace_back(cx, cy, orb);
+                    temp[index] = 1;
                 }
                 else
                 {
-                    // check left right
-                    hasOrbAround = hasSameOrb(orb, OrbLocation(cx, cy + 1)) && hasSameOrb(orb, OrbLocation(cx, cy - 1));
-                }
-
-                // only remove if there are no same orbs in a different direction
-                if (!hasOrbAround)
-                {
-                    board[INDEX_OF(cx, cy)] = pad::empty;
-                    list->emplace_back(cx, cy, orb);
+                    // shouldn't visit this orb again
+                    temp[index] = 2;
                 }
             }
+        }
 
-            for (int i = 0; i < currCount; i++)
+        for (int d = start; d < end; d++)
+        {
+            int startCount = -(dList[d * 2 + 1]);
+            int endCount = dList[d * 2] + 1;
+            for (int i = startCount; i < endCount; i++)
             {
                 int cx = x;
                 int cy = y;
                 if (d == 0)
                     cy += i;
                 else if (d == 1)
-                    cy -= i;
-                else if (d == 2)
                     cx += i;
-                else if (d == 3)
-                    cx -= i;
-
-                // fill all directions here
-                floodfill(list, OrbLocation(cx, cy + 1), orb, 0);
-                floodfill(list, OrbLocation(cx, cy - 1), orb, 1);
-                floodfill(list, OrbLocation(cx + 1, cy), orb, 2);
-                floodfill(list, OrbLocation(cx - 1, cy), orb, 3);
+
+                // prevent going to the same orb again
+                if (temp[INDEX_OF(cx, cy)] < 2)
+                {
+                    // fill all directions here
+                    floodfill(list, OrbLocation(cx + 1, cy), orb);
+                    floodfill(list, OrbLocation(cx - 1, cy), orb);
+                    floodfill(list, OrbLocation(cx, cy + 1), orb);
+                    floodfill(list, OrbLocation(cx, cy - 1), orb);
+                }
```

Comment:
Here, maybe we can check for `dList` because we know if there is a same orb in certain directions.